scapy/layers/dot11.py

```diff
@@ -640,11 +640,24 @@ def network_stats(self):
             elif isinstance(p, Dot11EltRates):
                 summary["rates"] = p.rates
             elif isinstance(p, Dot11EltRSN):
-                crypto.add("WPA2")
+                akmsuite_types = {
+                    0x01: "802.1X",
+                    0x02: "PSK"
+                }
+
+                auth = akmsuite_types.get(p.akm_suites[0].suite)
+
+                crypto.add("WPA2/%s" %(auth))
             elif p.ID == 221:
                 if isinstance(p, Dot11EltMicrosoftWPA) or \
                         p.info.startswith(b'\x00P\xf2\x01\x01\x00'):
-                    crypto.add("WPA")
+                    akmsuite_types = {
+                        0x01: "802.1X",
+                        0x02: "PSK"
+                    }
```

Comment:
Could you move the dict up in the function, so that it’s only defined once ?

Reply:
Also, it’s technically possible to have length = 0 (no akm_suites) available. You should check it’s not empty first

Reply:
Thanks for the suggestion, I have done.